tests/spicy/types/unit/skip.spicy

```diff
@@ -0,0 +1,38 @@
+# @TEST-EXEC: spicy-build -d -o test.hlto %INPUT
+
+# @TEST-EXEC: echo "Parsing X w/o any skipping" >> output
+# @TEST-EXEC: printf 'AB' | spicy-driver -p Test::X test.hlto >> output
+
+# @TEST-EXEC: echo "Parsing Y w/o any skipping" >> output
+# @TEST-EXEC: printf 'AB' | spicy-driver -p Test::Y test.hlto >> output
+
+# @TEST-EXEC: echo "Parsing X w/ content like skip-post" >> output
+# @TEST-EXEC: printf '01' | spicy-driver -p Test::X test.hlto >> output
+# @TEST-EXEC: printf '0011' | spicy-driver -p Test::X test.hlto >> output
+
+# @TEST-EXEC: echo "Parsing X w/o any skipping since skip-pre and -post in the wrong place" >> output
+# @TEST-EXEC: printf '1AB0' | spicy-driver -p Test::X test.hlto >> output
+# @TEST-EXEC: echo "Parsing Y w/o any skipping since skip-pre and -post in the wrong place" >> output
+# FIXME(bbannier): This is currently broken.
+# @_TEST-EXEC: printf '1AB0' | spicy-driver -p Test::Y test.hlto >> output
```

Comment:
Support for this is currently broken since we extract `X` values `['1', 'A', 'B']`. After that we still haven't reached EOD so the parser for `Y` kicks off parsing of another `X` value; that parse then `%skip-pre` drops the `0` and then has no more data to extract `&size=1` bytes.

In order for this to work we would either need to communicate some stop information from the parser of `X::x` up into the parser of `Y::xs`, or alternatively drop support for `%skip-pre` and `%skip-post` and let it all be handled by `%skip` (this would work since we could extract the `0` in a post step and then then detect EOD as usual).

Reply:
As we chatted about, I think these semantics are ok, we just need to document them.